openxla-nvgpu/compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNTypes.cpp

```diff
@@ -0,0 +1,47 @@
+//===- CUDNNTypes.cpp - CUDNN dialect types -----------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNTypes.h"
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/DialectImplementation.h"
+#include "openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNDialect.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace openxla::compiler::nvgpu::cudnn;
+
+static LogicalResult parseDimensionList(AsmParser &parser,
```

Comment:
I could have sworn we have this in mlir core already.

Reply:
Maybe. It was broken when I found it and I had to fix it up a bit due to upstream change.